packages/design-system/src/components/contextMenu/menu.js

```diff
@@ -17,379 +17,205 @@
-  content: '';
-  width: 100%;
-  height: 1px;
-  background-color: ${({ theme }) => theme.colors.divider.primary};
-  margin: 8px 0;
-
-  ${({ isIconMenu, theme }) =>
-    isIconMenu &&
-    css`
-      width: 40%;
-      margin: auto;
-      background-color: ${theme.colors.divider.primary};
-    `}
-`;
-separatorCSS.propTypes = {
-  isIconMenu: PropTypes.bool,
-};
-
-const MenuList = styled.ul`
-  background-color: ${({ theme }) => theme.colors.bg.primary};
-  border-radius: ${({ theme }) => theme.borders.radius.small};
-  margin: 0;
-  padding: ${({ isIconMenu }) => (isIconMenu ? 0 : '4px 0')};
-  pointer-events: auto;
-  list-style: none;
-
-  a {
-    background-color: transparent;
-    text-decoration: none;
+/**
+ * Finds all focusable children in an html tree.
+ *
+ * @param {Array.<HTMLElement>} all Array of html elements
+ * @param {HTMLElement} element The element to check
+ * @return {Array.<HTMLElement>} Array of html elements
+ */
+export function findFocusableChildren(all, element) {
+  // if element is group, check children and recurse
+  if (element.attributes?.role?.value === 'group' && element.children.length) {
+    return all.concat(
+      Array.from(element.children)
+        .map((child) => findFocusableChildren([], child))
+        .flat()
+    );
   }
-
-  li {
-    a,
-    button,
-    div {
-      display: flex;
-      justify-content: space-between;
-      align-items: center;
-      padding: 2px 16px;
-      border: 0;
-      color: ${({ theme }) => theme.colors.fg.primary};
-      transition: background-color ${BUTTON_TRANSITION_TIMING};
-    }
-    ${({ isIconMenu }) =>
-      isIconMenu &&
-      css`
-        margin: 0;
-        button {
-          padding: 0;
-          margin: 4px auto;
-        }
-        div,
-        a {
-          padding: 0;
-        }
-      `}
-
-    span {
-      transition: color ${BUTTON_TRANSITION_TIMING};
-    }
-
-    a {
-      :active span,
-      :hover span {
-        color: ${({ theme }) => theme.colors.fg.primary};
+  // if element is focusable, return element
+  // if element is not, return all
+  return FOCUSABLE_ELEMENTS.includes(element.tagName) && !element.disabled
+    ? all.concat(element)
+    : all;
+}
+
+const Menu = ({
+  children,
+  disableControlledTabNavigation,
+  isOpen,
+  onFocus = noop,
+  ...props
+}) => {
+  const { focusedId, isIconMenu, onDismiss, setFocusedId } = useContextMenu(
+    ({ state, actions }) => ({
+      focusedId: state.focusedId,
+      isIconMenu: state.isIconMenu,
+      onDismiss: actions.onDismiss,
+      setFocusedId: actions.setFocusedId,
+    })
+  );
+  const mouseDownOutsideRef = useMouseDownOutsideRef(() => {
+    isOpen && onDismiss?.();
+  });
+  const menuRef = useRef(null);
+  const composedListRef = useComposeRefs(mouseDownOutsideRef, menuRef);
+
+  const handleFocus = useCallback(
+    (evt) => {
+      onFocus(evt);
+
+      const focusableChildren = Array.from(
+        menuRef.current?.children || []
```

Comment:
💭 Have somewhere defined in the codebase that has all tags that are focusable. Just grab all focusable elements!